src/alert/alert.ts

```diff
@@ -51,9 +53,10 @@ export class NgbAlert {
  * Alerts that can be dismissed without any additional code.
  */
 @Directive({selector: 'template[ngbAlert]'})
-export class NgbDismissibleAlert implements OnInit {
+export class NgbDismissibleAlert implements OnInit, OnDestroy {
   private _windowFactory: ComponentFactory<NgbAlert>;
   private _windowRef: ComponentRef<NgbAlert>;
+  private _timeout: number;
```

Comment:
We shouldn't use `number` type here as what we get from `setTimeout` is a zone-specific data structure, not a number (`setTimeout` is patched by zone.js)